warp10/src/main/java/io/warp10/script/ext/token/TokenWarpScriptExtension.java

```diff
@@ -16,57 +16,73 @@
 
 package io.warp10.script.ext.token;
 
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Properties;
-
 import io.warp10.WarpConfig;
-import io.warp10.WarpDist;
 import io.warp10.crypto.KeyStore;
 import io.warp10.standalone.Warp;
 import io.warp10.warp.sdk.WarpScriptExtension;
 
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
 public class TokenWarpScriptExtension extends WarpScriptExtension {
-  
+
   /*
-   *  Name of configuration key with the token secret. 
+   * Name of configuration key with the token secret.
    */
   public static final String CONF_TOKEN_SECRET = "token.secret";
-  
+  /*
+   * Name of configuration key with OSS wrapped token secret
+   * ie: warp.key.token.secret  = wrapped:hex:xxxxx
+   */
+  public static final String KEY_TOKEN_SECRET = "warp.key.token.secret";
+
   /**
    * Current Token Secret
    */
   public static String TOKEN_SECRET = null;
-  
-  private static final Map<String,Object> functions = new HashMap<String,Object>();
+
+  private static final Map<String, Object> functions = new HashMap<String, Object>();
   private static final KeyStore keystore;
-  
+
   static {
     TOKEN_SECRET = WarpConfig.getProperty(CONF_TOKEN_SECRET);
-
-    if (null != TOKEN_SECRET) {
-      keystore = Warp.getKeyStore();
+    if (null == TOKEN_SECRET) {
+      // if OSS wrapped secret exists
+      TOKEN_SECRET = WarpConfig.getProperty(KEY_TOKEN_SECRET);
```

Comment:
Unwrapping of any configuration key prefixed with Configuration.WARP_KEY_PREFIX is automatic, the only thing you have to do is to attempt to retrieve the KEY_TOKEN_SECRET key and use it if it is set.

So KEY_TOKEN_SECRET should be "token.secret"

Reply:
indeed